### PR TITLE
Update authentication to use PAK

### PR DESCRIPTION
**Pull request by klucyszy**
Changes:
- API version upgrade from 1.15 to 1.22
- authentication method changed from ROPC to PAK
- avoid null reference errors when assigning lockProperties values

### CODE REVIEW

**Comment by NimlothPL** [comment]
Are U sure of this url:
api_url_base = "https://t01-nofo-api.azurewebsites.net/api/v1.22/"
?

Based on documentation it should be https://api.tedee.com/v1.22/

https://t01-nofo-api.azurewebsites.net/ looks like DEV as it have never API available (1.23 as of today) vs https://api.tedee.com (1.22)

**Comment by klucyszy** [comment]
@NimlothPL the link has been updated.

**Comment by NimlothPL** [comment]
Side note - I tested API with Personal Key and it works - nice :)
What should be documented (maybe in readme), minimal scope needed for this integration to work.
Also validTo is not well documented - exact format need to be used or also short one will work like "2025-11-05"

I tested with (manual API queries):
{
  "name": "Test",
  "validTo": "2021-11-05T15:33:29.054Z",
  "scopes": [
    "Device.Read",
    "Lock.Operate"
  ]
}

**Comment by klucyszy** [comment]
Hi, thanks for testing, and glad to hear that PAK is working fine.

I've added commit with an updated Readme file. Now it contains the minimum scopes required to use the integration. Regarding the date is the date format compliant with [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601). We will precise the Tedee documentation in the next iteration too.

Best regards
Kamil

File: pytedee/pytedee/TedeeClient.py [hunk 1]
```diff
@@ -194,51 +159,49 @@ def get_state(self):
                     if id == lock.get_id():
                         lock.set_connected(is_connected)
                         lockProperties = state["lockProperties"]
-                        _LOGGER.debug("Id: %s, State: %d, battery: %d", state["id"], lockProperties ["state"], lockProperties["batteryLevel"])
-                        battery_level = lockProperties["batteryLevel"]
-                        lock.set_battery_level(battery_level)
+                        lock_state = self.assign_null_or_lock_state(lockProperties)
+                        lock_batteryLevel = self.assign_null_or_lock_state(lockProperties)
```
Comment (joerg65) [inline pytedee/pytedee/TedeeClient.py]: Must be changed to:
```lock_batteryLevel = self.assign_null_or_lock_batteryLevel(lockProperties)```

File: pytedee/pytedee/TedeeClient.py [hunk 2]
```diff
@@ -194,51 +159,49 @@ def get_state(self):
...
-            self.get_token()
-        return self.is_token_valid() == True
-
     def find_lock(self, id):
         for lock in self._sensor_list:
             if id == lock.get_id():
                 return lock
         raise TedeeClientException("This Id not found")
-                      
-    def update(self, id):
```
Comment (joerg65) [inline pytedee/pytedee/TedeeClient.py:220]: For periodically update of the home assistant custom component, function update need to be there:
```
def update(self, id)
  self.get_state()
  return self.get_battery(id)
```

**Review by joerg65 — CHANGES_REQUESTED** [review]

**Comment by klucyszy** [comment]
@joerg65 thanks for review. I've resolved your comments. please check ;)

**Review by joerg65 — APPROVED** [review]

**Pull request merged**